x/qgb/types/ethereum_signer.go

```diff
@@ -18,40 +17,24 @@ func NewEthereumSignature(hash []byte, privateKey *ecdsa.PrivateKey) ([]byte, er
 	if privateKey == nil {
 		return nil, sdkerrors.Wrap(ErrEmpty, "private key")
 	}
-	protectedHash := crypto.Keccak256Hash(append([]uint8(signaturePrefix), hash...))
+	protectedHash := crypto.Keccak256Hash([]uint8(signaturePrefix), hash)
 	return crypto.Sign(protectedHash.Bytes(), privateKey)
 }
 
 func EthAddressFromSignature(hash []byte, signature []byte) (common.Address, error) {
 	if len(signature) < 65 {
 		return common.Address{}, sdkerrors.Wrap(ErrInvalid, "signature too short")
 	}
-	// To verify signature
-	// - use crypto.SigToPub to get the public key
-	// - use crypto.PubkeyToAddress to get the address
-	// - compare this to the address given.
-
-	// for backwards compatibility reasons  the V value of an Ethereum sig is presented
-	// as 27 or 28, internally though it should be a 0-3 value due to changed formats.
-	// It seems that go-ethereum expects this to be done before sigs actually reach it's
-	// internal validation functions. In order to comply with this requirement we check
-	// the sig an dif it's in standard format we correct it. If it's in go-ethereum's expected
-	// format already we make no changes.
-	//
-	// We could attempt to break or otherwise exit early on obviously invalid values for this
-	// byte, but that's a task best left to go-ethereum
-	if signature[64] == 27 || signature[64] == 28 {
-		signature[64] -= 27
```

Comment:
Out of curiosity, why is this no longer necessary?

Reply:
https://twitter.com/alexberegszaszi/status/1534461421454606336?t=6hUf4_GjhEuFBVdj_Kqzew&s=19

Reply:
wow so a bunch of smart contracts include this check even though it's unnecessary

Reply:
yes. And also in our case we were adding this check while it's not needed.